src/shared/bootspec.c

```diff
@@ -1525,42 +1520,38 @@ static int json_addon(
 static int json_cmdline(
                 const BootEntry *e,
                 const BootEntryAddons *global_arr,
+                const char *def_cmdline,
                 JsonVariant **v) {
 
-        _cleanup_free_ char *final_cmdline = NULL, *def_cmdline = NULL;
+        _cleanup_free_ char *combined_cmdline = NULL;
         _cleanup_(json_variant_unrefp) JsonVariant *addons_array = NULL;
         int r;
 
         assert(e);
 
-        if (!strv_isempty(e->options)) {
-                def_cmdline = strv_join(e->options, " ");
-                if (!def_cmdline)
-                        return log_oom();
-                final_cmdline = TAKE_PTR(def_cmdline);
-        }
+        if (def_cmdline)
+                combined_cmdline = strdup(def_cmdline);
 
-        FOREACH_ARRAY(addon, global_arr->items, global_arr->count) {
-                r = json_addon(addon, "globalAddon", &addons_array);
+        FOREACH_ARRAY(addon, global_arr->items, global_arr->n_items) {
+                r = json_addon(addon, "globaladdon", &addons_array);
```

Comment:
here you put together the JSON field. For that useing uppercasing is right, we do so already after all. hence globlAddon sounds good here.